lib/DataQueue.js

```diff
@@ -0,0 +1,221 @@
+const { ProgramCall } = require('./ProgramCall');
+const { CommandCall } = require('./CommandCall');
+// const { xmlToJson }   = require('./utils');
+
+// QRCVDTAQ Parameters
+const QRCVDTAQParameters = {
+  DataQueueName: { io: 'in',  type: '10A' },
+  LibraryName:   { io: 'in',  type: '10A' },
+  LengthOfData:  { io: 'out', type: '5p0' },
+  Data:          { io: 'out', type: '10A' },
+  WaitTime:      { io: 'in',  type: '5p0' }
+}
+
+class DataQueue {
+  /**
+   * @description Creates a new DataQueue object
+   * @constructor
+   * @param {object} connection
+   * @param {string} name
+   * @param {string} library
+   */
+  constructor(connection, name, library = '') {
+    this.connection = connection;
+
+    if (name.length > 10) {
+      throw Error("DataQueue name must be 10 or fewer characters long.");
+    }
+
+    if (library.length > 10) {
+      throw Error("DataQueue library must be 10 of fewer characers long.");
+    }
+
+    this.name = name;
+    this.library = library ? library : '*CURLIB';
+    this.qualifiedName = this.name.padEnd(10, ' ') + this.library.padEnd(10, ' ');
+    this.maxLength = undefined;
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Create Data Queue (CLDTAQ) CL Command
+  //////////////////////////////////////////////////////////////////////////////
+  /**
+   * @description Creates a Data Queue on the system of the connection passed in
+   *              the constructor.
+   * @constructor
+   * @param {number} maxLength
+   * @param {function} callback
+   */
+  async create(maxLength, callback) {
```

Comment:
Should these methods be defined as async? I'm thinking this could lead to synchronization issues. [Async function return a promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#return_value) and within the function we call `connection.run` with is asynchronous function using the callback pattern. So we wouldn't wait for `connection.run` to complete before returning the promise.

Reply:
So this goes with Issue #322 . I have always declared them as `async` since that's what they are. I wasn't aware the spec said I _had_ to return a Promise, just that "Async functions can contain zero or more await expressions." Because of the latter I don't think using callbacks internally is a big deal. Will have to dwell on it, but maybe is a reason people don't mix callbacks/Promises in the same function